### ISSUE TITLE
Please create a caller for php to append "async"

### ISSUE THREAD

**Issue by ttodua** [ccxt/ccxt#9593]
typically, we have to create a separate class for async version of exchange-classes. However, for portability, it would have been nice, that the single exchange class has been able to have a caller-callback for both versions. For example:
```
$exchange = new \ccxt\binance();
$exchange->fetch_ohlc();
```
but then we could also call:
`$exchange->async_fetch_ohlc();`
and that method called the "yield"-ed version. 

kinda annoying at this moment, having to create overloads for both classes and calling two different classes in our codes.

**Comment by kroitor** [comment]
> kinda annoying at this moment, having to create overloads for both classes and calling two different classes in our codes.

Can you please elaborate a bit? I'm having hard time imagining why would you not use just the async version or just the sync version of it, for all purposes. The underlying async framework actually allows you to set a callback on completion of a promise/future, so adding ad-hoc wrappers isn't that hard, but can be done outside of the library.

**Comment by ttodua** [comment]
Thanks for note. Actually, i tried that but having some problems (the `yield` keyword).

For example, in the application, we everywhere have syncronous functions. i.e.:

```
function AppFunc_1(){
    ...
    $res= AppFunc_2();
    ...
}

function AppFunc_2(){
    ...
    $ex = new \ccxt\async\bittrex();
    $barsData = $ex->fetch_ohlc();
    ...
}
```
`var_dump($barsData);`  returns generator object. If i use in that function:
```
    $barsData = yield $ex->fetch_ohlc();
```
Then the `AppFunc_2` becomes "generator" function for PHP, and thus all above hierarchy of application needs to be changed to handle 'yield'-ed function. Actually, I really wanted what you said initially, to just use 'async' version of ccxt and completely drop the sync-classes, but coulnd't get over that problem atm and after spending half day, i've given up... If you have any clues, I will be thankful.

**Comment by frosty00** [comment]
take a look at the asunc php examples:

https://github.com/ccxt/ccxt/tree/master/examples/async-php

**Comment by ttodua** [comment]
@frosty00  thanks for comment, but the examples are all done in plain .php file, with no namespace/class/method or nested method calls, so, i couldnt find out it (I've only recently started to find out what yield is and async in php), so i was unable to call generator (yield-based) functions in a normal method. I've tried to explain the exact example/issue in above comment - when I want to call "async" version of cctx, i have to use "$data=yield fetch_pair()"  which makes my method also "generator" kind. (and my method is being called from other parts of app, and so, all methods in chain needs to be turned into "generator" kind, which is impossible in our app. maybe i am missing something in php or dont know these tips, how to do it without forcing all methods in chain to be turned into "yiled/generator" kind.
